consoles/VNC.pm

```diff
@@ -921,6 +931,63 @@ sub _receive_zrle_encoding ($self, $x, $y, $w, $h) {
     return $res;
 }
 
+sub _receive_tight_encoding ($self, $x, $y, $w, $h) {
+    my $socket = $self->socket;
+
+    my $pi = $self->_pixinfo;
+
+    my $stime = time;
+    $socket->read(my $data, 1)
+      or OpenQA::Exception::VNCProtocolError->throw(error => 'short read for compression control');
+    my ($compression_control) = unpack('C', $data);
+    # FillCompression
+    if (($compression_control & 0xF0) eq 0x80) {
+        my $data;
+        # special case for TPIXEL, otherwise identical to PIXEL
+        if ($self->_true_colour and $self->_bpp eq 32 and $self->depth eq 24) {
+            $socket->read($data, 3)
+              or OpenQA::Exception::VNCProtocolError->throw(error => 'short read for compression control');
+            $data = pack('CCCx', unpack('CCC', $data));
+        } else {
+            $socket->read($data, $self->_bpp / 8)
+              or OpenQA::Exception::VNCProtocolError->throw(error => 'short read for compression control');
+        }
+        my $image = $self->_framebuffer;
+        $image->fill_pixel($data, $self->vncinfo, $x, $y, $w, $h);
+        return;
+    }
+    # Only Fill (above) and JPEG for now; "Basic" unsupported
+    die "Unsupported compression $compression_control" if ($compression_control & 0xF0) != 0x90;
+
+    $socket->read($data, 1)
+      or OpenQA::Exception::VNCProtocolError->throw(error => 'short read for data len');
+    my ($data_len) = unpack('C', $data);
+    if ($data_len & 0x80) {
+        $socket->read($data, 1)
+          or OpenQA::Exception::VNCProtocolError->throw(error => 'short read for data len');
+        my ($data_len2) = unpack('C', $data);
+        $data_len = ($data_len & 0x7f) | ($data_len2 & 0x7f) << 7;
+        if ($data_len2 & 0x80) {
+            $socket->read($data, 1)
+              or OpenQA::Exception::VNCProtocolError->throw(error => 'short read for data len');
+            my ($data_len2) = unpack('C', $data);
+            $data_len |= $data_len2 << 14;
+        }
+    }
+
+    $data = "";
+    my $read_len = 0;
+    while ($read_len < $data_len) {
+        my $len = $socket->read(my $data_tmp, $data_len - $read_len);
+        OpenQA::Exception::VNCProtocolError->throw(error => "short read for jpeg data $read_len - $data_len") unless $len;
+        $read_len += $len;
+        $data .= $data_tmp;
+    }
+    diag sprintf("read $data_len in %fs\n", time - $stime) if (time - $stime > 0.1);
```

Comment:
I'd avoid repeated calls to `time`:
```suggestion
    my $elapsed = time - $stime;
    diag "read $data_len in $elapsed\n" if $elapsed > 0.1;
```
But maybe that's just a leftover from debugging anyways?

Reply:
That's copied from ZRLE handling. But I guess it isn't really important.